surveys/2023-annual-survey/questions.md

```diff
@@ -0,0 +1,769 @@
+> We want to understand the "shape" of Rust usage, and this question tries to get at that
+> by allowing the respondent to select multiple answers.
+
+### Approximately how many total developers does your company employ?
+
+**Note**: don't worry about being exact here! Go with you gut.
+
+Type: select one (optional)
+
+- Under 10
+- 10-99
+- 100-1,000
+- 1,000-10,000
+- Over 10,000
+
+> This question is not that interesting on its own, but it can be used as a sort of cohort for understanding how answers
+> change depending on the size of the development effort at a company.
+>
+> Previously this question used "employees" instead of "developers". It is more appropriate for us to ask about the amount
+> of developers at a company vs. the amount of people employed in total.
+
+### Is your company planning on hiring Rust developers in the next year?
+
+Type: select one (optional)
+
+- Yes
+- No
+- I don't know
+
+> This question assess hiring sentiment. Although there is intrinsic uncertainty, it is easy to answer and forward looking.
+> It will also be interesting to see what the demand for Rust skills from companies is over time.
+
+## Your opinions about Rust
+
+### Which of the following statements about Rust do you feel are true?
+
+Type: select all that apply (optional)
+
+Statements:
+
+- Rust provides a real benefit over other programming languages
+- Rust is significantly more complicated to program in than other programming languages
+- Rust requires significantly more effort to learn than other programming languages
+- Rust code tends to contain significantly fewer bugs than equivalent code written in another programming language would have
+- Rust is risky to use in production
+- Rust makes me more productive
+- Rust is fun to use
+
+> **justification**
+>
+> There are several "truisms" about Rust that we'd like to get perspective on how true these are for users of Rust.
+>
+> Note that answers here can be subject to survivorship bias and so extra care should be taken with interpreting results.
+
+
+### What are your biggest worries for the future of Rust?
+
+Type: select all that apply (optional)
+
+- Not enough usage in industry
+- Too much interest from big companies
+- Not enough open source contributions to the ecosystem
+- Doesn't add a specific feature I want
+- Rust doesn't evolve quickly enough
+- Instability of the language
+- Superseded by an alternative
+- Becomes too complex
+- Tools and documentation are not accessible enough (e.g., due to language or incompatibility with screen readers)
+- Project governance does not scale to match the size/requirements of the community
+- Developers/maintainers of the language are not properly supported
+- I'm not worried
+- Other (open response)
+
+> **justification**
+>
+> Would be useful for leadership to understand the community's fears.
+
+### In your opinion, how would you prioritise work on the following aspects of Rust?
+
+Type: ordered list (optional)
+
+Aspects:
```

Comment:
What about other language features like variadic generics or an effect system?

Reply:
While I understand intuitively all the options, _effect system_ is not a term that I am familiar with prior to a lookup on Wikipedia or other resource (granted, it's a "hole" in my tech knowledge). Is there another way to formulate that in a way that is more approachable?

Reply:
I'm not sure if it's good to put specific features here. There are tens, if not hundreds of unfinished/nightly features that many people are eagerly expecting. I think that it would make more sense to create a new question asking specifically about the popularity of specific nightly features (although we can partially learn this from GitHub stats). We could list some of the most popular there, and let people include additional ones in an open form question.

Reply:
Yeah, I was thinking "new language features", optionally with some straightforward and popular examples.

Reply:
Ah, I see. Ok, so that's fine, I have added it.

But now that I have mentioned it, I think that it could be nice to actually have a separate question just for language features.

Reply:
Filed https://github.com/rust-lang/surveys/issues/238 for this.